pkg/trie/triedb/codec/node.go

```diff
@@ -52,21 +61,41 @@ type (
 	}
 )
 
-func (InlineValue) isNodeValue() {}
-func (HashedValue) isNodeValue() {}
+func (InlineValue) IsHashed() bool { return false }
+func (v InlineValue) Write(writer io.Writer) error {
+	encoder := scale.NewEncoder(writer)
+	err := encoder.Encode(v.Data)
+	if err != nil {
+		return fmt.Errorf("scale encoding storage value: %w", err)
+	}
+	return nil
+}
+
+func (HashedValue) IsHashed() bool { return true }
+func (v HashedValue) Write(writer io.Writer) error {
+	if len(v.Data) != common.HashLength {
+		panic("invalid hash length")
```

Comment:
Why panic and not an error?

Unless it is an error that would never happens (like safety check error, or an error that occurs only when we made some mistake in writing code), I feel we should not panick.